File: 118.cpp
class Solution {
public:
    vector<vector<int>> generate(int num_rows) {
        vector<vector<int>> result;

        result.push_back(std::vector<int>{1});

        for (int i = 1; i < num_rows; i++) {
            vector<int> curr(i + 1, 1);
            for (int j = 0; j <= i - 2; j++) {
                curr[j + 1] = result[i - 1][j] + result[i - 1][j + 1];
            }
            
            result.push_back(curr);
        }

        return result;
    }
};

File: 118.rs
impl Solution {
    pub fn generate(num_rows: i32) -> Vec<Vec<i32>> {
        let mut dp: Vec<Vec<i32>> = vec![];
        for i in 1..=num_rows {
            dp.push(vec![1; i as usize]);
        }

        for i in 0..num_rows as usize {
            if i == (num_rows - 1) as usize {
                break;
            }
            for j in 0..i as usize {
                dp[i + 1][j + 1] = dp[i][j] + dp[i][j + 1]; 
            }
        }

        dp
    }
}
